### PR TITLE
Add scripts to streamline dev env setup

### PR DESCRIPTION
**Pull request by yoomlam**
### Description
The Developer Setup instructions can be automated to expedite onboarding using the new docs/dev_env_setup_step*.sh scripts.

### Acceptance Criteria
- [X] Scripts work in a terminal on a Mac operating systems.

### Testing Plan
1. New onboarders should test the scripts on their new Macbooks.

### Documentation Updates
- [X] "Developer Setup" section in README.md updated

### CODE REVIEW

**Comment by pkarman** [comment]
Great to see these. Maybe they belong in `scripts/` instead of `docs/` though?

File: docs/dev_env_setup_step2.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# Continuation of Developer Setup at
```
Comment (alisan16) [inline docs/dev_env_setup_step2.sh]: Is there a reason why this part of the setup needs to be separated into a second script?

Comment (yoomlam) [reply docs/dev_env_setup_step2.sh]: Done.

File: README.md [hunk 2]
```diff
@@ -109,6 +109,15 @@ We are using 2-factor authentication with Github so, for example, when you acces
 
 ### Machine setup #######################################################
 
+You can manually go through the following steps.
+Alternatively, if you have a Mac, you can download and run the following scripts:
+
+  - [dev_env_setup_step1.sh](docs/dev_env_setup_step1.sh)
```
[the remark this answers is not in the document]
Comment (yoomlam) [reply README.md]: Done.

**Comment by yoomlam** [comment]
This PR went off my radar. I moved the scripts to address the comments, so it should be ready.

**Pull request merged**